pageserver/src/virtual_file/owned_buffers_io/write/flush.rs

```diff
@@ -0,0 +1,323 @@
+    /// The submitted buffer has not been flushed to disk.
+    #[cfg(test)]
+    NotStarted(FlushNotStarted),
+}
+
+/// A control object that manipulates buffer's flush behehavior.
+pub(crate) struct FlushControl {
+    state: FlushStartState,
+}
+
+impl FlushControl {
+    #[cfg(test)]
+    fn not_started(
+        ready_to_flush_tx: tokio::sync::oneshot::Sender<()>,
+        done_flush_rx: tokio::sync::oneshot::Receiver<()>,
+    ) -> Self {
+        FlushControl {
+            state: FlushStartState::NotStarted(FlushNotStarted {
+                ready_to_flush_tx,
+                done_flush_rx,
+            }),
+        }
+    }
+
+    #[cfg(not(test))]
+    fn untracked() -> Self {
+        FlushControl {
+            state: FlushStartState::Untracked,
+        }
+    }
+
+    /// In tests, turn flush control into a not started state.
+    #[cfg(test)]
+    pub(crate) fn into_not_started(self) -> FlushNotStarted {
+        match self.state {
+            FlushStartState::NotStarted(not_started) => not_started,
+        }
+    }
+
+    /// Release control to the submitted buffer.
+    ///
+    /// In `cfg(test)` environment, the buffer is guranteed to be flushed to disk after [`FlushControl::release`] is finishes execution.
+    pub async fn release(self) {
+        match self.state {
+            #[cfg(not(test))]
+            FlushStartState::Untracked => (),
+            #[cfg(test)]
+            FlushStartState::NotStarted(not_started) => {
+                not_started
+                    .ready_to_flush()
+                    .wait_until_flush_is_done()
+                    .await;
+            }
+        }
+    }
+}
+
+impl<Buf, W> FlushHandle<Buf, W>
+where
+    Buf: IoBufAligned + Send + Sync + Clone,
+    W: OwnedAsyncWriter + Send + Sync + 'static + std::fmt::Debug,
+{
+    /// Spawns a new background flush task and obtains a handle.
+    ///
+    /// Note: The background task so we do not need to explicitly maintain a queue of buffers.
+    pub fn spawn_new<B>(file: Arc<W>, buf: B, ctx: RequestContext) -> Self
+    where
+        B: Buffer<IoBuf = Buf> + Send + 'static,
+    {
+        let (front, back) = duplex::mpsc::channel(2);
+
+        let join_handle = tokio::spawn(async move {
+            FlushBackgroundTask::new(back, file, ctx)
+                .run(buf.flush())
+                .await
+        });
+
+        FlushHandle {
+            inner: Some(FlushHandleInner {
+                channel: front,
+                join_handle,
+            }),
+            maybe_flushed: None,
+        }
+    }
+
+    /// Submits a buffer to be flushed in the background task.
+    /// Returns a buffer that completed flushing for re-use, length reset to 0, capacity unchanged.
+    /// If `save_buf_for_read` is true, then we save the buffer in `Self::maybe_flushed`, otherwise
+    /// clear `maybe_flushed`.
+    pub async fn flush<B>(
+        &mut self,
+        buf: B,
+        offset: u64,
+        save_buf_for_read: bool,
```

Comment:
ONce the `write_buffered` method is gone, there's no need for this parameter anymore